sdk/scheduler/src/test/java/com/mesosphere/sdk/api/JettyApiServerTest.java

```diff
@@ -75,4 +83,22 @@ public Response getFrameworkId() {
             return Response.ok(TEST, MediaType.APPLICATION_JSON).build();
         }
     }
+
+    private void sleep(int ms) {
+        // an interruptable sleep b/c if interrupted, i.e. by SIGINT, np,
+        // continue to crash, don't worry about sleep
+        try {
+            Thread.sleep(ms);
+        } catch (InterruptedException e) { }
+    }
+
+    private int getJettyApiServerPort() {
+        if (jettyApiServer == null) {
```

Comment:
This seems a little weird. I'd just throw a runtime exception here.

Or, just call `jettyApiServer.getLocalPort()` directly at https://github.com/mesosphere/dcos-commons/pull/992/files#diff-c9f51af3b06d6f233439266a232d6d3cR72

I think that would fail in a logical both cases?

Right now, you have to see what request failed to parse, then see what that number means in this method.